pkg/envoy/cell.go

```diff
@@ -187,3 +212,78 @@ func newArtifactCopier(lifecycle hive.Lifecycle) *ArtifactCopier {
 
 	return artifactCopier
 }
+
+type syncerParams struct {
+	cell.In
+
+	Logger      logrus.FieldLogger
+	Lifecycle   hive.Lifecycle
+	JobRegistry job.Registry
+	Scope       cell.Scope
+
+	K8sClientset client.Clientset
+
+	Config    secretSyncConfig
+	XdsServer XDSServer
+}
+
+func registerSecretSyncer(params syncerParams) error {
+	if !params.Config.EnableIngressController && !params.Config.EnableGatewayAPI {
+		return nil
+	}
+
+	secretSyncer := newSecretSyncer(params.Logger, params.XdsServer)
+
+	// Create a Secret Resource for each namespace.
+	// The Cilium Agent only has permissions on the specific namespaces.
+	// Note that the different features can use the same namespace for
+	// their TLS.
+	namespaces := map[string]struct{}{}
+
+	for namespace, cond := range map[string]func() bool{
+		params.Config.EnvoySecretsNamespace:      func() bool { return option.Config.EnableEnvoyConfig },
+		params.Config.IngressSecretsNamespace:    func() bool { return params.Config.EnableIngressController },
+		params.Config.GatewayAPISecretsNamespace: func() bool { return params.Config.EnableGatewayAPI },
+	} {
+		if cond() {
+			namespaces[namespace] = struct{}{}
```

Comment:
Do we need to check if ns is empty?

Reply:
good point. i added an additional check that the namespace isn't of zero length.